x-pack/plugins/ml/server/models/job_validation/validate_cardinality.ts

```diff
@@ -4,21 +4,22 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import _ from 'lodash';
-
+import { APICaller } from 'kibana/server';
 import { DataVisualizer } from '../data_visualizer';
 
 import { validateJobObject } from './validate_job_object';
+import { CombinedJob } from '../../../common/types/anomaly_detection_jobs';
+import { Detector } from '../../../common/types/anomaly_detection_jobs';
 
-function isValidCategorizationConfig(job, fieldName) {
+function isValidCategorizationConfig(job: CombinedJob, fieldName: string): boolean {
   return (
     typeof job.analysis_config.categorization_field_name !== 'undefined' &&
     fieldName === 'mlcategory'
   );
 }
 
-function isScriptField(job, fieldName) {
-  const scriptFields = Object.keys(_.get(job, 'datafeed_config.script_fields', {}));
+function isScriptField(job: CombinedJob, fieldName: string): boolean {
+  const scriptFields = Object.keys(job?.datafeed_config?.script_fields ?? {});
```

Comment:
nit, it doesn't look like `job` will be undefined here, `datafeed_config` also shouldn't be undefined.

Reply:
Done in https://github.com/elastic/kibana/pull/65971/commits/90ba7a4ca3e8ba42a52d1afcb76d79aaf5a161b2